transaction/atomic.go

```diff
@@ -54,18 +54,17 @@ func getAccountDetails(hezClient client.HermezClient, address string,
 	return
 }
 
-// AtomicTransfer perform token or ETH transfers in a pool of transactions
-func AtomicTransfer(hezClient client.HermezClient, ethereumChainID int,
-	txs []AtomicTxItem) (serverResponse string, err error) {
-	atomicGroup := api.AtomicGroup{}
-
+// CreateFullTxs turn the basic information in a PoolL2Tx, set metadata and fields based on the current state. Also
+// links the txs setting the Rq* fields.
+func CreateFullTxs(hezClient client.HermezClient, txs []AtomicTxItem) (fullTxs []hezcommon.PoolL2Tx, err error) {
 	// configure transactions and do basic validations
 	for currentAtomicTxId := range txs {
 		localTx := hezcommon.PoolL2Tx{}
+		hezcommon.NewPoolL2Tx(&localTx)
```

Comment:
this should be called after seting the tx fields (Amount, nonce, ....), since they're usedd to set the `TxID` and `Type`